North-Seattle-College/ad440-winter2021-thursday-repo
# GitHub Action to Test users/user_id/tests/test_id

### Description
Resolves #132

**This PR creates an Action that on merge, will test endpoint 'users/user_id/tests/test_id' and then store the results to the 'artillery' s3 bucket.**

### Added features:
- Workflow file for creating the action
- Shell script for s3 set up
- Python script for s3 file upload with boto3

### Screenshots:
**Action results you should see**
![Screen Shot 2021-02-16 at 7 26 44 PM](https://user-images.githubusercontent.com/52220118/108152035-fbf80580-708c-11eb-8acb-3fd069448fc2.png)

### To Test:
**- This script needs to be merged in order to fully test due to issues with AWS credential privileges. Therefore partial testing will be done in a separate repo.**
**- Also the action may return errors depending on the state of the endpoint.**

If you plan to review, please shoot me a slack message so I can add you as a collaborator for this test repo. Or you could try forking and testing on your own (I think that should work as well).
1. Visit my test repo and $ git clone https://github.com/jleeboyd/artilleryActionTest
2. Now add something like a blank file or random comment and push up to the repo (push protections not in place)
3. Check out the actions https://github.com/jleeboyd/artilleryActionTest/actions and check to make sure you reached the same step 'Upload to s3', with the same error message 'InvalidAccessKey' (Should match screenshot above)
   - **Note: The action errors out because of AWS privileges with our secrets. Which is why this PR needs to be merged with the NSC repo before it can be fully tested.**
4. If you saw the error message in 3. then the action is working as expected.

**Note:** Once this action has been merged, results should be uploaded to the AWS 'artillery' S3 bucket with the following example naming convention: Ex. artillery/2021/02/08/20-26-22.100.json. Your file will be the date and time your artillery test ran, which will be in the **Run Artillery Test step in the GitHub Action** [Artillery Bucket](https://s3.console.aws.amazon.com/s3/buckets/artillery?region=us-west-2&tab=objects).

Time Spent
|**DATE**|**ACTIVITY**|**TIME**|
|---|:---|:---|
|2/9|Identify task requirements by re-watching lecture and looking at previous actions|1 hours|
|2/10|Break down task and begin research into steps|3.5 hours|
|2/15|Begin coding out the workflow|2 hour|
|2/16|Finish coding, and incorporate s3 upload|5 hour|

Total time: 11.5 hr
